test/verifiedTests/design/testAnalyzeGCdesign.m

```diff
@@ -0,0 +1,45 @@
+% The COBRAToolbox: testAnalyzeGCdesign.m
+%
+% Purpose:
+%     - test the analyzeGCdesign function
+%
+% Authors:
+%     - Jacek Wachowiak
+global CBTDIR
+% save the current path
+currentDir = pwd;
+
+% initialize the test
+fileDir = fileparts(which('testAnalyzeGCdesign'));
+cd(fileDir);
+
+% test variables
+model = readCbModel([CBTDIR filesep 'test' filesep 'models' filesep 'ecoli_core_model.mat']);
+selectedRxns = model.rxns(1);
+target = model.rxns(2);
+deletions = model.rxns(3);
+modelRed = reduceModel(model);
+
+% function outputs
+% requires Global Optimization Toolbox
+% function operates on a not full rank matrix and therefore cannot end without an error
+try
+    [improvedRxns, intermediateSlns] = analyzeGCdesign(modelRed, selectedRxns, target, deletions);
+catch ME
+    assert(length(ME.message) > 0)
+end
+try
+    [improvedRxns, intermediateSlns] = analyzeGCdesign(modelRed, selectedRxns, target, '');
+catch ME
+    assert(length(ME.message) > 0)
+end
+for i=2:9
+  try
+      [improvedRxns, intermediateSlns] = analyzeGCdesign(modelRed, selectedRxns, target, deletions, i, i);
+  catch ME
+      assert(length(ME.message) > 0)
+  end
+end
+
```

Comment:
Shouldn't a test function test whether the output of a function is correct and what would be expected?
while yes, testing errors is also sensible, if the only test is whether the function fails it would be better to remove the function.

Reply:
I tried in different ways to obtain a positive result but since the function can succeed only if it is given a full rank matrix (that should be created by the other functions it calls) it always throws an error. Changing input arguments - different models, adding reduceModel gave no results.

Reply:
Admittedly I'm currently at a loss as to what this function is supposed to do bc the documentation is, well, "interesting". 
I think the first thing for this function would be additional documentation of the expected inputs/outputs.
My current guess is:
Based on a given set of present deletions, find a set of deletions that optimizes the flux through the specified target reaction. To do so, either reactions are removed from the list, or reactions from the "selectedRxns" are replacing one reactions from the current list, or reactions from selected Rxns are being added to the current list.

So, if you want to use this on ecoli core as a test, I would suggest to use e.g. EX_ac(e) as target (improve Acetate production), while setting a few deletions that do not influence the Biomass production.